src/client/rest/RESTMethods.js

```diff
@@ -31,72 +34,73 @@ class RESTMethods {
           content = content.replace(/@(everyone|here)/g, '@\u200b$1');
         }
 
         if (split) content = splitMessage(content, typeof split === 'object' ? split : {});
       }
 
+      const send = (chan) => {
+        if (content instanceof Array) {
+          const messages = [];
+          (function sendChunk(list, index) {
+            const options = index === list.length ? { tts, embed } : { tts };
+            chan.send(list[index], options, index === list.length ? file : null).then((message) => {
+              messages.push(message);
+              if (index >= list.length) return resolve(messages);
+              return sendChunk(list, ++index);
+            });
+          }(content, 0));
+        } else {
+          this.rest.makeRequest('post', Constants.Endpoints.channelMessages(chan.id), true, {
+            content, tts, nonce, embed,
+          }, file).then(data => resolve(this.client.actions.MessageCreate.handle(data).message), reject);
+        }
+      };
+
       if (channel instanceof User || channel instanceof GuildMember) {
-        this.createDM(channel).then(chan => {
-          this._sendMessageRequest(chan, content, file, tts, nonce, embed, resolve, reject);
-        }, reject);
+        this.createDM(channel).then(send, reject);
       } else {
-        this._sendMessageRequest(channel, content, file, tts, nonce, embed, resolve, reject);
+        send(channel);
       }
     });
   }
 
-  _sendMessageRequest(channel, content, file, tts, nonce, embed, resolve, reject) {
-    if (content instanceof Array) {
-      const datas = [];
-      let promise = this.rest.makeRequest('post', Constants.Endpoints.channelMessages(channel.id), true, {
-        content: content[0], tts, nonce,
-      }, file).catch(reject);
-
-      for (let i = 1; i <= content.length; i++) {
-        if (i < content.length) {
-          const i2 = i;
-          promise = promise.then(data => {
-            datas.push(data);
-            return this.rest.makeRequest('post', Constants.Endpoints.channelMessages(channel.id), true, {
-              content: content[i2], tts, nonce, embed,
-            }, file);
-          }, reject);
-        } else {
-          promise.then(data => {
-            datas.push(data);
-            resolve(this.rest.client.actions.MessageCreate.handle(datas).messages);
-          }, reject);
-        }
-      }
-    } else {
-      this.rest.makeRequest('post', Constants.Endpoints.channelMessages(channel.id), true, {
-        content, tts, nonce, embed,
-      }, file)
-        .then(data => resolve(this.rest.client.actions.MessageCreate.handle(data).message), reject);
+  updateMessage(message, content, { embed, code } = {}) {
+    content = this.client.resolver.resolveString(content);
+    if (code) {
+      content = escapeMarkdown(this.client.resolver.resolveString(content), true);
+      content = `\`\`\`${typeof code === 'string' ? code : ''}\n${content}\n\`\`\``;
```

Comment:
Use `typeof code !== 'undefined' && code !== null`, so that objects and numbers will be stringified, which is more consistent with how passing non-strings as strings typically behaves.